### DISCUSSION TITLE
Adding icalendar to thanks.dev - funding

### DISCUSSION THREAD

**Discussion by niccokunzmann** [collective/icalendar#512] [Ideas]
https://thanks.dev is a dependency funding platform. While Github Sponsors and Open Collective fund single projects, this one funds all dependencies. I would like to try it out to add icalendar to thanks.dev and wanted to start a discussion here as to how to approach it as this is also within the Plone community.

Open questions:
- [ ] Where does the money go? 
    - Should not be much at the start.
    - We can pass it to our dependencies for example.
    - Should it go to Plone?

I had a call with Armin from thanks.dev and we can have one together. I think, thanks.dev are interested in having some projects like this because a lot of projects depend on this one. I already added mine because funding FOSS is an interest of mine at the moment.

What are your thoughts on this?
@geier @mauritsvanrees @jacadzaca and the others who have contributed in the last months? And anyone reading this.

**Comment by niccokunzmann** [comment]
I also informed the Plone community about this discussion: https://community.plone.org/t/future-of-icalendar-looking-for-maintainer/15445/3?u=niccokunzmann

**Comment by jacadzaca** [comment]
Hi @niccokunzmann, thanks.dev looks promising. To my knowledge, icalendar's only dependencies are pytest, tox and sphinx, but these probably aren't on thanks.dev, so from what I understand we wouldn't be able to pass the donations to them, right?

How about using the potential funds we'd gather for setting up a bounty system for the pending issues? Would that be possible to integrate with Github? Would accepting 'targeted donations' from people who'd *really* want to see an issue resolved be implementable?

**Reply by niccokunzmann** [reply]
thanks.dev can pass the funds to opencollective for example. I can set up an open collective with you and the other maintainers. Then, we can use the money incoming to award it to issues. I cannot remember where but I have seen an issue that had some dollars in the title. I think it might have been weblate. I will have a look. That would be a possible place to fund issues. Looking ...

Yes, example how they do it: https://github.com/WeblateOrg/weblate/issues/3106

We can probably use the same system.
I care about the aliveness of the project. If money adds to people coming and contributing that is alright. If they stay, then for other reasons, usually.

What do you think? Should we try?

**Reply by niccokunzmann** [reply]
It seems like this to me:

1. People use thanks.dev to donate to their dependencies
2. money goes to the icalendar collective - funds are publicly viewable and where they go (Example from open-web-calendar: https://opencollective.com/open-web-calendar )
3. We can send them to bountysource (Example from weblate: https://app.bountysource.com/issues/81891384-markdown-files-support )
    - [ ] TODO: How does that integrate with PayPal - that seems to be the only option on bountysource
4. We choose an issue to fund.
5. Once the issue is solved, by some rules of bounty-source, the money is released.

I think, this is a very transparent process in every step.
If people fund the project, I think, they would expect something like this to go on.
So, this looks like a nice thing to try out.
I do not see any problems. 
At the start the amounts will probably be small if there are any. This is what I see for my project which accounts for about 1/4 of icalendar usage:

> No donors found for your projects yet.
> Hang in there – we're working on it!

I was on a call with one of the founders of thanks.dev. I think, I will post the next one here, too, so you can join in if you like. I hope, they will focus on FOSS and not so much on the network effect that can be used to make it big.

So, I think, it is nice to try it out, no money will flow in the next months either way. So, there is plenty of time to get prepared. 
I will be waiting a little for answers from the Plone community.

What are your thoughts on this?

**Reply by mauritsvanrees** [reply]
I can't spend enough time to form a real opinion, but thanks.dev sounds interesting, so I am happy for you to try something out.

**Reply by niccokunzmann** [reply]
So, I went ahead and created a collective.

This is the roadmap that I see:

- [x] Create a collective: https://opencollective.com/python-icalendar/
- [x] Await approval - DONE. richard[at]oscollective.org is our goto person.
- [ ] Add one more icalendar maintainer as admin to the collective @jacadzaca @mauritsvanrees @geier and also @angatha  and @stlaz if you like to help us with funding. Please register at the platform if you like to support this experiment and give me your username so I can add you. The purpose is to decentralize administration. If you like to support later, that is also possible.
- [ ] Register at thanks.dev and add the collective as target
- [x] schedule a call with thanks.dev of how to further cooperate
- [ ] wait for some support money
- [ ] create an account on bountysource
- [ ] agree on an issue to fund
- [ ] get the issue solved
- [ ] transfer some money
- [ ] Celebrate that this process worked out to solve an issue :fireworks: :tada: 

Would you add something to that?

**Reply by anehzat** [reply]
> It seems like this to me:
> 
> 1. People use thanks.dev to donate to their dependencies
> 2. money goes to the icalendar collective - funds are publicly viewable and where they go (Example from open-web-calendar: https://opencollective.com/open-web-calendar )
> 3. We can send them to bountysource (Example from weblate: https://app.bountysource.com/issues/81891384-markdown-files-support )
>    
>    * [ ]  TODO: How does that integrate with PayPal - that seems to be the only option on bountysource
> 4. We choose an issue to fund.
> 5. Once the issue is solved, by some rules of bounty-source, the money is released.
> 
> I think, this is a very transparent process in every step. If people fund the project, I think, they would expect something like this to go on. So, this looks like a nice thing to try out. I do not see any problems. At the start the amounts will probably be small if there are any. This is what I see for my project which accounts for about 1/4 of icalendar usage:
> 
> > No donors found for your projects yet.
> > Hang in there – we're working on it!
> 
> I was on a call with one of the founders of thanks.dev. I think, I will post the next one here, too, so you can join in if you like. I hope, they will focus on FOSS and not so much on the network effect that can be used to make it big.
> 
> So, I think, it is nice to try it out, no money will flow in the next months either way. So, there is plenty of time to get prepared. I will be waiting a little for answers from the Plone community.
> 
> What are your thoughts on this?

BTW. We are working with Paypal to see if they can work something out as a way for us to better support open source maintainers.

**Reply by niccokunzmann** [reply]
I scheduled a call with thanks.dev on Sunday, 16th April at 22:00 UTC. If you like to partake, send me an email.

**Reply by niccokunzmann** [reply]
In order to fund work on single issues, I requested that https://polar.sh has access to write to the issues.
See the [example](https://github.com/niccokunzmann/open-web-calendar/issues/288):
![image](https://github.com/collective/icalendar/assets/564768/5eac7a53-1627-4eba-93fc-bba77a6cb934)

Using polar.sh, we can actually allow people to click on a funding button below the issues.

**Reply by mauritsvanrees** [reply]
I have approved the request to allow polar.sh on the collective/icalendar repo.

polar.sh is now "Connecting your amazing repositories", which it has been doing for several minutes now, so I wonder if it is stalled or is doing too much.

In my own settings on GitHub I see:

> polar.sh can access your account mauritsvanrees to:
> 
> Verify your GitHub identity
> Know what resources you can access
> Act on your behalf
> View your email addresses
>
> polar.sh has been installed on 1 account you have access to: collective.

That sounds like I gave it access to my own account as well, so maybe I did too much.
But in the settings of the collective org it only has access to the icalendar repo, so seen from icalendar it should be fine this way.

**Reply by niccokunzmann** [reply]
I opened this issue: https://github.com/polarsource/polar/issues/2938
There, you can see that I do not have access to edit the org. I wonder: On the top-left, can you choose icalendar as an option to configure?
